### ISSUE TITLE
[19.07, master] Asterisk 16.x is not build with ASLR PIE

### ISSUE THREAD

**Issue by BKPepe** [openwrt/telephony#502]
Maintainer: @jslachta 

When PKG_ASLR_PIE is enabled for the build Asterisk 16.x can not be compiled due to this error. Most probably similar to [#9845](https://github.com/openwrt/packages/pull/9845) from packages feed.

Build log is same for OpenWrt 19.07 and OpenWrt master:
```
/-----/build/staging_dir/toolchain-arm_cortex-a9+vfpv3_gcc-7.5.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/7.5.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld: ast_expr2fz.o: relocation R_ARM_MOVW_ABS_NC against `stdin' can not be used when making a shared object; recompile with -fPIC
ast_expr2fz.o: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Makefile:182: recipe for target 'check_expr2' failed
make[4]: *** [check_expr2] Error 1
```
By default PKG_ASLR_PIE is in the default settings disabled and it is possible that it can be enabled in the future. 

As workaround till the issue is find and fixed, `PKG_ASLR_PIE:=0` should be applied to the Makefile.

**Comment by micmac1** [comment]
Could this be a bug in binutils? Then I'd prefer not to hide it in Asterisk package.

**Comment by dhewg** [comment]
untested, but it sounds like `TARGET_CFLAGS+=$(FPIC)` could fix it?

**Comment by micmac1** [comment]
That'd be bad, very bad. This is an application, not a lib.

**Comment by BKPepe** [comment]
Do you think that this can be included also in the branch `openwrt-19.07`?

**Comment by micmac1** [comment]
On Sat, Mar 07, 2020 at 01:23:30AM -0800, Josef Schlehofer wrote:
> Do you think that this can be included also in the branch `openwrt-19.07`?

Hi Josef,

Can you cherry-pick the commit and send a pull request? I don't think
Jiri would be opposed to this.

Kind regards,
Seb
